### PR TITLE
2/3 Add check in / check out endpoints

### PR DESCRIPTION
**Pull request by seantcanavan**
1. Implement `CancelCheckInAppointment`
2. Implement `CheckInAppointment`
3. Implement `CheckOutAppointment`
4. Implement `GetRequiredCheckInFields`
5. Implement `StartCheckInAppointment`

@dillonstreator @fterrag need advice on the breaking change to `BookAppointment`. I originally naively implemented the input arguments as integers however the API tends to return strings so to make functions more easily chainable I modified the input args to be strings. While integration testing I found myself needlessly converting a previous function's output string value to the `BookAppointment` function's integer parameter. If we want this breaking change to be in a separate PR or to be reverted just let me know. We're fine with this breaking change on the CopilotIQ side since we haven't gone to prod yet but wanted to triple check with y'all first.

Screenshot of copilotiq integration test success included since athena is so gracious to incorrectly document their return values.
Screenshot from 2023-07-06 01-11-33

### CODE REVIEW

**Comment by fterrag** [comment]
@seantcanavan I'm no longer at Eleanor Health, but @dillonstreator or @elliehastings will take care of you. Thanks for contributing to this project!

File: athenahealth/client.go [hunk 1]
```diff
@@ -24,9 +24,15 @@ type Client interface {
...
 	ListOpenAppointmentSlots(ctx context.Context, departmentID int, opts *ListOpenAppointmentSlotOptions) (*ListOpenAppointmentSlotsResult, error)
-	BookAppointment(ctx context.Context, patientID, apptID int, opts *BookAppointmentOptions) (*BookedAppointment, error)
+	BookAppointment(ctx context.Context, patientID, apptID string, opts *BookAppointmentOptions) (*BookedAppointment, error)
 	RescheduleAppointment(ctx context.Context, apptID int, opts *RescheduleAppointmentOptions) (*RescheduleAppointmentResult, error)
 
+	CancelCheckInAppointment(ctx context.Context, apptID string) error
+	CheckInAppointment(ctx context.Context, apptID string) error
+	CheckOutAppointment(ctx context.Context, apptID string) error
+	GetRequiredCheckInFields(ctx context.Context, deptID string) (*GetRequiredCheckInFieldsResult, error)
+	StartCheckInAppointment(ctx context.Context, apptID string) error
```
[the remark this answers is not in the document]
Comment (seantcanavan) [reply athenahealth/client.go:34]: no worries - i'm actually going to move GetRequiredFields to department.go since it's a department call

**Comment by seantcanavan** [comment]
changes made

**Comment by seantcanavan** [comment]
@dillonstreator @elliehastings any chance for a review here? got a few PRs lined up after this one. thanks! :heart:

**Review by dillonstreator — CHANGES_REQUESTED** [review]

File: athenahealth/insurance.go [hunk 2]
```diff
@@ -84,18 +84,11 @@ func (h *HTTPClient) CreatePatientInsurancePackage(ctx context.Context, opts *Cr
...
-}
-
 // ReactivatePatientInsurancePackage - Reactivate a patient's insurance package
-//
 // POST /v1/{practiceid}/patients/{patientid}/insurances/{insuranceid}/reactivate
-//
 // https://docs.athenahealth.com/api/api-ref/patient-insurance#Reactivate-patient's-specific-insurance-package
 func (h *HTTPClient) ReactivatePatientInsurancePackage(ctx context.Context, patientID, insuranceID string, expirationDate *time.Time) error {
-	out := &reactivatePatientInsurancePackageResponse{}
+	out := &ErrorMessageResponse{}
```
[the remark this answers is not in the document]
Comment (seantcanavan) [reply athenahealth/insurance.go]: fixed!

**Comment by seantcanavan** [comment]
@dillonstreator thanks! ready for re-review :nerd_face:

**Pull request merged**